code/modules/mob/living/carbon/human/human_mob.dm

```diff
@@ -795,8 +795,31 @@
 			if(!modified)
 				to_chat(usr, "<span class='warning'>Unable to locate a data core entry for this person.</span>")
 
+	if(href_list["mental"])
+		if(hasHUD(usr, EXAMINE_HUD_MEDICAL_WRITE))
+			if(usr.incapacitated())
+				return
+			var/modified = 0
+			var/perpname = get_visible_name(TRUE)
+
+			for(var/datum/data/record/E in GLOB.data_core.general)
+				if(E.fields["name"] == perpname)
+					for(var/datum/data/record/R in GLOB.data_core.general)
+						if(R.fields["id"] == E.fields["id"])
+							var/setmental = input(usr, "Specify a new mental status for this person.", "Medical HUD", R.fields["m_stat"]) in list("*Insane*", "*Unstable*", "*Watch*", "Stable", "Cancel")
+
+							if(hasHUD(usr, EXAMINE_HUD_MEDICAL_WRITE))
+								if(setmental != "Cancel")
+									R.fields["m_stat"] = setmental
+									modified = 1
+									spawn()
+										sec_hud_set_security_status()
```

Comment:
Is this supposed to reference the security status here?

Reply:
judging by the medical lines of code just above, yes

Reply:
after examining the code, probably not
```dm
/mob/living/carbon/human/proc/sec_hud_set_security_status()
	var/image/holder = hud_list[WANTED_HUD]
	var/perpname = get_visible_name(TRUE) //gets the name of the perp, works if they have an id or if their face is uncovered
	if(!SSticker) return //wait till the game starts or the monkeys runtime....
	if(perpname)
		var/datum/data/record/R = find_record("name", perpname, GLOB.data_core.security)
		if(R)
			switch(R.fields["criminal"])
				if(SEC_RECORD_STATUS_EXECUTE)
					holder.icon_state = "hudexecute"
					return
				if(SEC_RECORD_STATUS_ARREST)
					holder.icon_state = "hudwanted"
					return
				if(SEC_RECORD_STATUS_SEARCH)
					holder.icon_state = "hudsearch"
					return
				if(SEC_RECORD_STATUS_MONITOR)
					holder.icon_state = "hudmonitor"
					return
				if(SEC_RECORD_STATUS_DEMOTE)
					holder.icon_state = "huddemote"
					return
				if(SEC_RECORD_STATUS_INCARCERATED)
					holder.icon_state = "hudincarcerated"
					return
				if(SEC_RECORD_STATUS_PAROLLED)
					holder.icon_state = "hudparolled"
					return
				if(SEC_RECORD_STATUS_RELEASED)
					holder.icon_state = "hudreleased"
					return
	holder.icon_state = null
```